students/RoyC/Lesson05/mailroom.py

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env python3
+# Lesson 5, Mailroom with exception handling and Comprehension
+
+# Donor data dictionary
+donors = {
+    "Ned Flanders": [1200.25, 850.35],
+    "Martin Prince": [12.22, 19.56],
+    "Edna Krabappel": [55.43, 118.67, 75.23],
+    "Homer Simpson": [253.64, 772.50, 99.99],
+    "Moe Szylak": [54.23]
+}
+
+# template used to format thank you letter
+ltr_template = ("\n\nDear {donor_name},"
+                "\n\nThank you for your generous donation of ${amt}."
+                "\nThis brings your to-date total of donations to ${total}!"
+                "\nYour kind help is greatly appreciated."
+                "\n\nKindest regards, Monty Burns\n\n"
+               )
+
+def send_thanx():
+    """
+    Prompt operator for donor name, or 'list' to see names of donors on record.
+    Take name entered and either use existing donor, or add new record.
+    Then prompt for amount of this donation, then save the donation record and
+    print a thank you note.
+    """
+    while True:
+        donor_name = input("\nEnter full name of donor (or 'list' to see names of donors): ")
+        if donor_name == "list":
+            for d in donors:
+                print(d)
+            print()
+        else:
+            amt = 0.0
+            donations = donors.setdefault(donor_name, [])
+            while True:
+                try:
+                    amt = float(input("Enter amount of donation: "))
+                except ValueError:
+                    print("Not a valid donation format, please enter a number")
+                else:
+                    break
+            donations.append(amt)
+            record = {'donor_name' : donor_name, 'amt' : "{:.2f}".format(amt), 'total' : "{:.2f}".format(get_total(donations))}
+            print(ltr_template.format(**record))
+            break
+    
+def thank_all():
+    """
+    Write thank you letters for all donors to separate files
+    """
+    for donor, donations in donors.items():
+        f = open(donor.replace(' ',  '_') + ".txt", 'w')
+        last_donation = donations[-1:][0]
+        donor_record = {'donor_name' : donor, 'amt' : "{:.2f}".format(last_donation), 'total' : "{:.2f}".format(get_total(donations))}
+        f.write(ltr_template.format(**donor_record))
+    print("\n{:d} thank you letters have been written!\n".format(len(donors)))
+
+def get_total(donations):
+    """
+    Return the total of the given list of donations
+    Argument:
+        donations - list of donation amounts (floating point nums)
+    """
+    total = 0;
+    for donation in donations:
+        total += donation
+    return total
```

Comment:
the whole thing can be replaced with `sum` function... this was also a good candidate for comprehensions (which are part of this lesson).